src/lib/helpers/youtubePlayerHandling.ts

```diff
@@ -95,6 +98,12 @@ export const youtubePlayerParsing = async (
                             .decipher(
                                 innertubeClient.session.player,
                             );
+                    videoData.streamingData.adaptiveFormats[index].url =
+                        videoData.streamingData.adaptiveFormats[index].url
+                            .replace(
+                                /pot=[^&]*/,
+                                `pot=${innertubeClient.session.po_token}`,
+                            );
```

Comment:
I'm thinking of swapping to parsing and reconstructing this URL rather than trying to regex replace - it's getting messy. I'll do that when I next look at this, or I can do it in a refactor PR after.

Reply:
"pot=" parameter is not returned by YouTube server in the adaptiveFormats, so I'm surprised that you have to do such thing.

The example implementation just append the pot parameter at the end of the url: https://github.com/LuanRT/BgUtils/blob/main/examples/node/innertube-challenge-fetcher-example.ts#L102

Reply:
ah - I missed cleaning this bit up actually. It doesn't need to be there at all - they actually do have a PO token (at least in dash?) and it works as-is
![image](https://github.com/user-attachments/assets/d4340026-ac81-4ad0-869a-7f4037f9ee80)